llvm/lib/Transforms/Utils/SCCPSolver.cpp

```diff
@@ -232,6 +234,33 @@ static bool replaceSignedInst(SCCPSolver &Solver,
   return true;
 }
 
+/// Try to use \p Inst's value range from \p Solver to simplify it.
+static Value *simplifyInstruction(SCCPSolver &Solver,
+                                  SmallPtrSetImpl<Value *> &InsertedValues,
+                                  Instruction &Inst) {
+  auto GetRange = [&Solver, &InsertedValues](Value *Op) {
```

Comment:
We should probably unify this, the same code is also used in `refineInstruction` and `replaceSignedInst` in a slightly different variant.